### PR TITLE
Allow image quality to be defined through the url and to be taken int…

### PR DESCRIPTION
**Pull request by moddular**
…o account when calculating etags

### CODE REVIEW

File: lib/immagine/service.rb [hunk 1]
```diff
@@ -38,50 +38,72 @@ class Service < Sinatra::Base
...
+      setup_image_processing(dir, format_code, quality, basename)
+
+      source_file = source_file_path(dir, basename)
+
+      set_etag_and_cache_headers(dir, format_code, quality, basename, source_file)
+      generate_image(format_code, quality, source_file, convert_to: newformat.to_sym)
+    end
+
+    def image_quality(quality = nil)
+      Integer(quality || request.env['HTTP_X_IMAGE_QUALITY'] || DEFAULT_IMAGE_QUALITY)
```
Comment (dazoakley) [inline lib/immagine/service.rb]: Remove the header?  I don't think we ever used it...

**Comment by dazoakley** [comment]
Looks good 👍 

But might be worth giving the README some love soon - there's a _lot_ of new functionality in there that just isn't documented.

File: lib/immagine/service.rb (hunk 1 above)
Comment (moddular) [inline lib/immagine/service.rb]: Will do. Just fixing up the hound errors, I'll remove that too

**Comment by dazoakley** [comment]
👍

**Pull request merged**